### PR TITLE
sstable: do not rename if tempdir is not specified

### PR DESCRIPTION
**Pull request by tchaikov**
before this change, we already rename the component file if it is opened for write, even if the temp_dir is not set. in that case, the underlying rename() syscall is a noop, but it still a waste.

so, instead of checking `readonly`, we just check if the `tgt_dir` is the sstable's dir. this is not only more readable, but also more correct in the sense it avoid unnecessary syscall.

### CODE REVIEW

**Comment by tchaikov** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2106/

minio server failed somehow.

**Comment by tchaikov** [comment]
@xemul hi Pavel, could you please help review this change?

File: sstables/storage.cc [hunk 1]
```diff
@@ -111,16 +111,15 @@ future<file> filesystem_storage::open_component(const sstable& sst, component_ty
     auto name = sst.filename(tgt_dir, type);
 
     auto f = open_sstable_component_file_non_checked(name, flags, options, check_integrity);
+    if (tgt_dir == dir) {
```
Comment (xemul) [inline sstables/storage.cc:114]: This looks like slower check for me -- it's effectively the strcmp. While it was a bool-flag check. The `temp_dir` can be unset only in some rare corner cases (if at all), so I'd keep the existing check even if it loses some cycles in false-positive case.

Comment (tchaikov) [reply sstables/storage.cc:114]: okay. i don't want to insist. but the `strcmp` versus bool-flag check is not the key here. the point is that `tgt_dir == dir` is more correct.

**Pull request closed without merging**